core/geometry/src/geometry3d/Ray3d.ts

```diff
@@ -54,20 +61,50 @@ export class Ray3d implements BeJSONFunctions {
     }
     return new Ray3d(Point3d.createZero(), Vector3d.createZero());
   }
-  /** Test for nearly equal Ray3d objects.
+  /**
+   * Test for nearly equal Ray3d objects.
    * * This tests for near equality of origin and direction -- i.e. member-by-member comparison.
-   * * Use [[isAlmostEqualPointSet]] to allow origins to be anywhere along the common ray and to have to allow the directions to be scaled or opposing.
-  */
+   * * Use [[isAlmostEqualPointSet]] to allow origins to be anywhere along the common ray and to have to allow the
+   * directions to be scaled or opposing.
+   */
   public isAlmostEqual(other: Ray3d): boolean {
     return this.origin.isAlmostEqual(other.origin) && this.direction.isAlmostEqual(other.direction);
   }
-  /** Test for nearly equal rays, allowing origin float and direction scaling.
+  /** Return the dot product of the ray's direction vector with a vector from the ray origin to the `spacePoint`. */
+  public dotProductToPoint(spacePoint: Point3d): number {
+    return this.direction.dotProductStartEnd(this.origin, spacePoint);
+  }
+  /** Return the fractional coordinate (along the direction vector) of the `spacePoint` projected to the ray. */
+  public pointToFraction(spacePoint: Point3d): number {
+    return Geometry.safeDivideFraction(
+      this.dotProductToPoint(spacePoint),
+      this.direction.magnitudeSquared(),
+      0
+    );
+  }
+  /** Return the `spacePoint` projected onto the ray. */
+  public projectPointToRay(spacePoint: Point3d): Point3d {
+    /**
+     * To project a point to the ray, we can create a vector called "v" from ray origin to the spacePoint and project
+     * that vector to the ray "r". The projection is "((v.r)/||r||^2) r" where "v.r" is the dort product. Note that
+     * pointToFraction returns "(v.r)/||r||^2".
+     * Note: If ray is the normal of a plane, then projection length "(v.r)/||r||" is the altitude of the spacePoint
+     * with respect to the plane.
+     */
+    return this.origin.plusScaled(this.direction, this.pointToFraction(spacePoint));
+  }
+  /**
+   * Test for nearly equal rays, allowing origin float and direction scaling.
    * * Use [[isAlmostEqual]] to require member-by-member comparison.
-  */
+   */
   public isAlmostEqualPointSet(other: Ray3d): boolean {
+    /**
+     * This function tests two rays to determine if they define the same infinite lines.
+     * So the origins can be different as long as they are on the infinite line (they can
+     * "float") but the directions must be parallel or antiparallel.
+     */
     if (!this.direction.isParallelTo(other.direction, true))
       return false;
-    // In exact math, it is not possible for one origin to be on the other ray but not vice versa.  But we'll test both ways.
```

Comment:
I'd leave this comment in, as it explains the apparent redundancy below (which is prudent given that we ae not "in exact math").